zuko/transforms.py

```diff
@@ -633,17 +633,12 @@ class BernsteinTransform(MonotonicTransform):
     bijective = True
     sign = +1
 
-    def __init__(
-        self, theta: Tensor, smooth_bounds: bool = True, keep_in_bounds: bool = False, **kwargs
-    ):
+    def __init__(self, theta: Tensor, **kwargs):
         super().__init__(None, phi=(theta,), **kwargs)
 
-        self.theta = self._constrain_theta(
-            theta, smooth_bounds, self.bound if keep_in_bounds else None
-        )
-        self.keep_in_bounds = keep_in_bounds
-
+        self.theta = self._constrain_theta(theta)
         self.order = self.theta.shape[-1] - 1
+
         dtheta = self.order * (self.theta[..., 1:] - self.theta[..., :-1])
 
         self.basis = self._bernstein_basis(self.order, device=theta.device, dtype=theta.dtype)
```

Comment:
Is this part necessary for the bounded transform? I guess that offsets should be $-B$ and $B$ and that slopes should be $1$? Or maybe there are numerical errors?

Also if you don't want to expand the dimensions of `x`, you can evaluate the Bernstein polynomial(s) at `self.eps` and `1 - self.eps` separately. Scalars always broadcast!

Reply:
Good point. I refactored the calculations into a method and overload it with one returning fixed values in the bounded subclass.